src/main/surveys/surveyTriggerManager.js

```diff
@@ -14,102 +14,316 @@
-            func: "{surveyTriggerManager}.handleRuleSuccess",
+        onTriggerOccurred: {
+            func: "{that}.removeTrigger",
             args: [
-                "{arguments}.0", // triggerType
-                "{arguments}.1"  // payload
+                "{arguments}.0" // trigger
             ]
         }
-
-    },
-    components: {
-        rulesEngine: null
     },
-
     invokers: {
-        reset: {
-            funcName: "gpii.app.surveyTriggerManager.reset",
-            args: "{that}"
-        },
-
-        handleRuleSuccess: {
-            funcName: "gpii.app.surveyTriggerManager.handleRuleSuccess",
+        registerTrigger: {
+            funcName: "gpii.app.surveyTriggerManager.registerTrigger",
             args: [
                 "{that}",
-                "{arguments}.0", // triggerType
-                "{arguments}.1"  // payload
+                "{arguments}.0" // trigger
             ]
         },
-
-        registerTrigger: {
-            funcName: "gpii.app.surveyTriggerManager.registerTrigger",
+        removeTrigger: {
+            funcName: "gpii.app.surveyTriggerManager.removeTrigger",
             args: [
-                "{that}.options.triggerTypes.surveyTrigger",
-                "{rulesEngine}",
-                "{arguments}.0" // triggerData
+                "{that}",
+                "{arguments}.0" // trigger
             ]
+        },
+        reset: {
+            funcName: "gpii.app.surveyTriggerManager.reset",
+            args: ["{that}"]
         }
     }
 });
 
 /**
- * Clears any registered triggers.
- * @param that {Component} The `gpii.app.surveyTriggerManger` component
+ * Registers a survey trigger with the `surveyTriggerManager` by creating a
+ * trigger handler for it. Any trigger handlers for previously registered
+ * triggers that have the same id as the current trigger will be destroyed
+ * and will no longer be tracked.
+ * @param that {Component} The `gpii.app.surveyTriggerManager` instance.
+ * @param trigger {Object} The survey trigger which is to be registered.
  */
-gpii.app.surveyTriggerManager.reset = function (that) {
-    var triggerTypes = fluid.values(that.options.triggerTypes);
+gpii.app.surveyTriggerManager.registerTrigger = function (that, trigger) {
+    that.removeTrigger(trigger);
+    that.events.onTriggerAdded.fire(trigger);
+};
+
+/**
+ * Removes a trigger from the `surveyTriggerManager` by destroying its
+ * corresponding trigger handler.
+ * @param that {Component} The `gpii.app.surveyTriggerManager` instance.
+ * @param trigger {Object} The survey trigger which is to be removed.
+ */
+gpii.app.surveyTriggerManager.removeTrigger = function (that, trigger) {
+    var triggerId = trigger.id;
+    if (fluid.isValue(triggerId)) {
+        var triggerHandler = that.registeredTriggerHandlers[triggerId];
+        if (triggerHandler) {
+            triggerHandler.destroy();
+            delete that.registeredTriggerHandlers[triggerId];
```

Comment:
Unnecessary, destroying the component will automatically unlink it from the tree

Reply:
My idea here was to explicitly remove the reference to the destroyed component so that it can be garbage collected and thus to avoid any potential memory leaks.